src/cpp/src/sampler.cpp

```diff
@@ -587,45 +590,95 @@ SamplerOutput Sampler::sample(std::vector<SequenceGroup::Ptr> & sequence_groups,
 
         const void * sequence_group_logits_data = logits_data + vocab_size * currently_processed_tokens;
         ov::Tensor sequence_group_logits(ov::element::f32, ov::Shape{num_running_sequences, actual_seq_len, vocab_size}, (void *)sequence_group_logits_data);
-
+        size_t decrease_len = 0;
         if (sequence_group->requires_sampling()) {
+            auto token_id = actual_seq_len;
+            // prompt phase
+            if (sequence_group->get_num_processed_tokens() == 0) {
```

Comment:
in case of chunked prefill, prompt can be processed via multiple iterations.
And it seems we will have overflow below after subtraction

Reply:
It's handled inside the condition. I have updated the code just to improve readability